src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/ConnectionString.cs

```diff
@@ -0,0 +1,308 @@
+            });
+            
+            if(nonEmptyUriList.IsAny<KeyValuePair<string,string>>())
+            {
+                nonEmptyUriKeyName = nonEmptyUriList.FirstOrDefault().Key;
+            }
+
+            return nonEmptyUriKeyName;
+        }
+
+        /// <summary>
+        /// Detects if Environment was set in the connection string
+        /// </summary>
+        /// <returns>True: If valid environment was set. False:If environment was empty or invalid</returns>
+        private bool IsEnvironmentSet()
+        {
+            bool envSet = false;            
+            string envNameString = KeyValuePairs.GetValueUsingCaseInsensitiveKey(ConnectionStringKeys.EnvironmentKey);
+            if (!string.IsNullOrEmpty(envNameString))
+            {
+                EnvironmentNames envName;
+                if (!Enum.TryParse<EnvironmentNames>(envNameString, out envName))
+                {
+                    string envError = string.Format("Environment '{0}' is not valid. Possible values:'{1}'", envNameString, envName.ListValues());
+                    ParseErrors = envError;
+                    throw new ArgumentException(envError);
+                }
+
+                envSet = !string.IsNullOrEmpty(envName.ToString());
+            }
+
+            return envSet;
+        }
+       
+        #endregion
+
+        #region Public Functions
+        /// <summary>
+        /// Parses connection string
+        /// </summary>
+        /// <param name="connString">Semicolon delimented KeyValue pair(e.g. KeyName1=value1;KeyName2=value2;KeyName3=value3)</param>
+        public void Parse(string connString)
+        {
+            string keyName;
+            string parseRegEx = @"(?<KeyName>[^=]+)=(?<KeyValue>.+)";
+
+            if (_parseErrorSb != null) _parseErrorSb.Clear();
+
+            if (string.IsNullOrEmpty(connString))
+            {
+                ParseErrors = "Empty connection string";
+            }
+            else
+            {
+                string[] pairs = connString.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
+                if (pairs == null) ParseErrors = string.Format("'{0}' unable to parse string", connString);
+
+                //TODO: Shall we clear keyValue dictionary?
+                //What if parsed gets called on the same instance multiple times
+                //the connectionString is either malformed/invalid
+                //For now clearing keyValue dictionary, we assume the caller wants to parse new connection string
+                //and wants to discard old values (even if they are valid)
+
+                KeyValuePairs.Clear(clearValuesOnly: true);
+                foreach (string pair in pairs)
+                {
+                    Match m = Regex.Match(pair, parseRegEx);
+
+                    if (m.Groups.Count > 2)
+                    {
+                        keyName = m.Groups["KeyName"].Value.ToLower();
```

Comment:
if we make this a case-insensitive dictionary, we shoule not have to do any case manipulation